### PR TITLE
Mapbox custom visual area selection(lasso) on LineString

### PR DESCRIPTION
**Pull request by tenshi13**
jira: https://intelehub.atlassian.net/browse/DN-880

tested: 
- on chrome 87.0.4280.88
image

- on firefox 84.0.2 (64-bit)
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mapboxgl-powerbi",
-  "version": "20201201",
+  "version": "1.5.0",
```
Comment (tenshi13) [inline package.json:3]: this is actually describing the node version needed to compile the code, reverted to 1.5.0 as per original mapboxgl

File: pbiviz.json [hunk 2]
```diff
@@ -1,18 +1,18 @@
 {
   "visual": {
-    "name": "mapboxGLMap",
-    "displayName": "Mapbox Visual",
+    "name": "Intelematics - mapboxGLMap",
+    "displayName": "Intelematics - Mapbox Visual",
     "guid": "PBI_CV_EB3A4088_75C5_4746_9D8B_255A7B7ECD6D",
     "visualClassName": "MapboxMap",
-    "version": "20201201",
+    "version": "20200113-1",
```
Comment (tenshi13) [inline pbiviz.json:7]: this we can change and should be used to track our internal version of this pbiviz

File: pbiviz.json [hunk 3]
```diff
@@ -1,18 +1,18 @@
 {
   "visual": {
-    "name": "mapboxGLMap",
-    "displayName": "Mapbox Visual",
+    "name": "Intelematics - mapboxGLMap",
+    "displayName": "Intelematics - Mapbox Visual",
```
Comment (tenshi13) [inline pbiviz.json:4]: added intelematics to differentiate our pbiviz to original mapboxgl pbiviz

File: src/build/turf-build.js [hunk 4]
```diff
@@ -6,5 +6,6 @@ module.exports = {
     center: require('@turf/center').default,
     booleanContains: require('@turf/boolean-contains').default,
     booleanOverlap: require('@turf/boolean-overlap').default,
+    lineIntersect: require('@turf/line-intersect').default,
```
Comment (tenshi13) [inline src/build/turf-build.js:9]: need to at drawControl.ts -> selectFeature function, see below

File: src/ts/drawControl.ts [hunk 5]
```diff
@@ -36,9 +36,10 @@ module powerbi.extensibility.visual {
                     if (feature.geometry.type === 'Point' && turf.booleanContains(sel_pol, feature)) {
                         return true;
                     }
-                    if ((feature.geometry.type === 'Polygon' || feature.geometry.type === 'Linestring') &&
-                       (turf.booleanOverlap(feature, sel_pol) || turf.booleanContains(sel_pol, feature) ||
-                        turf.booleanContains(feature, sel_pol)
+                    if (
+                        (feature.geometry.type === 'Polygon' || feature.geometry.type === 'LineString') &&
+                        (turf.lineIntersect(feature, sel_pol) || turf.booleanContains(sel_pol, feature) ||
```
Comment (tenshi13) [inline src/ts/drawControl.ts:41]: booleanOverlap can only compare same feature types, e.g. polygon vs polygon or lineString vs lineString, lineIntersect supports cross feature type compare so used this here instead

File: .gitignore [hunk 6]
```diff
@@ -61,6 +61,7 @@ typings/
 # Compiled files
 src/ts/*.js*
 src/ts/**/*.js*
+src/js/turf-bundle.js
```
Comment (tenshi13) [inline .gitignore:64]: this is intermediate file generated by `npm run turf-build` no need to be git tracked

**Review by john-intele — APPROVED** [review]

File: pbiviz.json (hunk 2 above)
Comment (thai-intelematics) [reply pbiviz.json:7]: The year is so 2020

**Pull request merged**